.github/workflows/integration-test.yml

```diff
@@ -1292,28 +1292,32 @@ jobs:
           terraform init
           if terraform apply --auto-approve \
             -var="beta=true" \
-            -var="addon_name=amazon-cloudwatch-observability" \
-            -var="addon_version=v1.6.0-eksbuild.1" \
-            -var="k8s_version=1.29" ; then
+            -var="addon_name=${{ matrix.arrays.addon_name }}" \
+            -var="addon_version=${{ matrix.arrays.addon_version }}" \
+            -var="ami_type=${{ matrix.arrays.ami }}" \
+            -var="instance_type=${{ matrix.arrays.instanceType }}" \
+            -var="k8s_version=${{ matrix.arrays.k8s_version }}"; then
             echo "Terraform apply successful."
           
             # Capture the output
             echo "Getting EKS cluster name"
             EKS_CLUSTER_NAME=$(terraform output -raw eks_cluster_name)
             echo "Cluster name is ${EKS_CLUSTER_NAME}"
-            kubectl apply -f ./gpuBurner.yaml
-            kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v0.15.0/deployments/static/nvidia-device-plugin.yml
+            kubectl create -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v0.17.0/deployments/static/nvidia-device-plugin.yml
             kubectl patch amazoncloudwatchagents -n amazon-cloudwatch cloudwatch-agent --type='json' -p='[{"op": "replace", "path": "/spec/image", "value": ${{ secrets.AWS_ECR_PRIVATE_REGISTRY }}/${{ env.ECR_INTEGRATION_TEST_REPO }}:${{ github.sha }}}]'
+            echo "Sleep to wait for device plugin pods to start"
+            sleep 10
```

Comment:
instead of using sleep can we do 
```
kubectl wait --for=condition=Ready pod --all -n kube-system
kubectl wait --for=condition=Ready pod --all -n amazon-cloudwatch
```